plugin/trino-kudu/src/main/java/io/trino/plugin/kudu/KuduRecordSet.java

```diff
@@ -70,7 +72,10 @@ public RecordCursor cursor()
 
     KuduTable getTable()
     {
-        return kuduSplit.getTableHandle().getTable(clientSession);
+        if (kuduTable == null) {
+            kuduTable = clientSession.openTable(kuduSplit.getSchemaTableName());
```

Comment:
can we call `openTable` more times with this refactor. If so is it a problem?

Reply:
`KuduRecordSet` is created once per split by `KuduRecordSetProvider#getRecordSet`. Caching the `kuduTable` on the `KuduRecordSet` should be functionally equivalent of caching it at the `KuduSplit` level.